thirdeye/thirdeye-frontend/app/pods/rootcause/controller.js

```diff
@@ -246,43 +273,51 @@ export default Ember.Controller.extend({
       return;
     }
 
-    // NOTE: potential race condition?
     [...missing].forEach(urn => pending.add(urn));
 
     this.setProperties({_pendingTimeseriesRequests: pending});
     this.notifyPropertyChange('_pendingTimeseriesRequests');
 
-    console.log('_startRequestMissingTimeseries: set pending', pending);
+    // metrics
+    const metricUrns = [...missing].filter(urn => urn.startsWith('thirdeye:metric:'));
+    const metricIdString = metricUrns.map(urn => urn.split(":")[2]).join(',');
+    const metricUrl = `/timeseries/query?metricIds=${metricIdString}&ranges=${context.analysisRange[0]}:${context.analysisRange[1]}&granularity=15_MINUTES&transformations=timestamp`;
+
+    fetch(metricUrl)
+      .then(res => res.json())
+      .then(this._extractTimeseries)
+      .then(timeseries => this._completeRequestMissingTimeseries(this, timeseries));
 
-    const metricIds = [...missing].map(urn => urn.split(":")[2]);
+    // baselines
+    const baselineOffset = context.anomalyRange[0] - context.baselineRange[0];
+    const baselineDisplayStart = context.analysisRange[0] - baselineOffset;
+    const baselineDisplayEnd = context.analysisRange[1] - baselineOffset;
 
-    const idString = metricIds.join(',');
-    const url = `/timeseries/query?metricIds=${idString}&ranges=${context.analysisRange[0]}:${context.analysisRange[1]}&granularity=15_MINUTES&transformations=timestamp,relative`;
+    const baselineUrns = [...missing].filter(urn => urn.startsWith('frontend:baseline:metric:'));
+    const baselineIdString = baselineUrns.map(urn => urn.split(":")[3]).join(',');
+    const baselineUrl = `/timeseries/query?metricIds=${baselineIdString}&ranges=${baselineDisplayStart}:${baselineDisplayEnd}&granularity=15_MINUTES&transformations=timestamp`;
 
-    fetch(url)
+    fetch(baselineUrl)
       .then(res => res.json())
-      .then(this._extractTimeseries)
-      .then(incoming => this._completeRequestMissingTimeseries(this, incoming));
+      .then(res => this._extractTimeseries(res))
+      .then(timeseries => this._convertMetricToBaseline(timeseries, baselineOffset))
+      .then(timeseries => this._completeRequestMissingTimeseries(this, timeseries));
 
   },
 
   _completeRequestMissingTimeseries(that, incoming) {
```

Comment:
Also the use of arrow functions should let us avoid passing the context. In which case is this failing?

Reply:
^ This is the only blocking comment

Reply:
semi-fix. arrow context works for entities loading, but not for timeseries loading. this is a strange case.